src/Microsoft.Framework.DependencyInjection/ServiceLookup/ServiceScope.cs

```diff
@@ -8,20 +8,37 @@ namespace Microsoft.Framework.DependencyInjection.ServiceLookup
     internal class ServiceScope : IServiceScope
     {
         private readonly ServiceProvider _scopedProvider;
+        private readonly ServiceScopeFactory _parentFactory;
 
         public ServiceScope(ServiceProvider scopedProvider)
         {
             _scopedProvider = scopedProvider;
         }
+        internal ServiceScope(ServiceProvider scopedProvider, ServiceScopeFactory parentFactory)
+        {
+            _scopedProvider = scopedProvider;
+            _parentFactory = parentFactory;
+        }
 
         public IServiceProvider ServiceProvider
         {
             get { return _scopedProvider; }
         }
 
+        internal void Reset()
+        {
+            // Double wash
```

Comment:
Phrased another way, original dispose is to drop any references so the can be GC'ed as the object is now kept; second Reset dispose is to make it safe for reuse.

I could swap ```_disposables``` to a backing object to make any adds error; then restore it in the Reset instead?